js/source/worker.js

```diff
@@ -47,7 +48,22 @@ util.extend(Worker.prototype, {
             if (err) return callback(err);
 
             tile.data = new vt.VectorTile(new Protobuf(new Uint8Array(data)));
-            tile.parse(tile.data, this.layers, this.actor, callback);
+
+            // if a parentTile is defined it means the index is missing a tile for this coord
+            // here the difference between the requested tile and its indexed parent is found
+            // we pass the dz, x/y pos of the tile's relationship to its parent
+            if (params.parentId && tile.data.layers) {
+                var tilePos = TileCoord.fromID(params.coord.id);
+                var parentPos = TileCoord.fromID(params.parentId);
+                var dz = tilePos.z - parentPos.z;
+                var xPos = tilePos.x & ((1 << dz) - 1);
+                var yPos = tilePos.y & ((1 << dz) - 1);
+
+                // chelm - i'd prefer to not just tack on params here...
+                tile.parse(tile.data, this.layers, this.actor, callback, dz, xPos, yPos);
```

Comment:
Passing dz, and x/y position offset to the `worker_tile.parse` method. I'd like a cleaner way to do this, passing 7 params to a function is no bueno.